### PR TITLE
Add support for abbreviated titles in direct_html

### PR DESCRIPTION
**Pull request by nik9000**
In docbook we can abbreviate the title of a section in the table of
contents by doing something like this:

```
== Intro to XYZ
++++
<titleabbrev>Intro</titleabbrev>
++++

Words.
```

This is *very* docbook. `++++` mean "pass this output directly to the
generated doc" and `<titleabbrev>` is a docbook element.

This change adds support for abbreviating a title of a section in the
table of contents when using `--direct_html` using the following syntax:

```
[id=intro_to_xyz,titleabbrev=Intro]
== Intro to XYZ

Words.
```

It is both shorter and more AsciiDoc friendly. It isn't compatible with
docbook though.

This change *also* adds support for identifying the docbook-style title
abbreviate commands when running with `--direct_html` and having those
set the title abbreviation. In other words: you don't have to change the
old books, but new books that use `--direct_html` should probably use
the new syntax.

### CODE REVIEW

File: README.asciidoc [hunk 1]
```diff
@@ -338,17 +338,38 @@ book.  See <<linking>>.
...
+one of two ways:
 
+. If the doc is built with `--direct_html` you *should*
+add a `titleabbrev` attribute to the section:
++
+--
 [source,asciidoc]
-----------------------------------
+----
+[id=intro_to_xyz,titleabbrev=Intro]
```
Comment (jrodewig) [inline README.asciidoc]: I would do an example that includes multiple words and a space.

```suggestion
[id=intro_to_xyz,titleabbrev=" XYZ Intro"]
```

File: README.asciidoc [hunk 2]
```diff
@@ -338,17 +338,38 @@ book.  See <<linking>>.
...
+a pass block with `<titleabbrev>`. If the doc is built
+with `--direct_html` then you *may* use the pass block
+but it isn't recommended:
++
+--
+[source,asciidoc]
+----
+== Intro to XYZ
 ++++
 <titleabbrev>Intro</titleabbrev>
```
Comment (jrodewig) [inline README.asciidoc]: Just an update to match the earlier example.

```suggestion
<titleabbrev>XYZ Intro</titleabbrev>
```

File: README.asciidoc [hunk 3]
```diff
@@ -338,17 +338,38 @@ book.  See <<linking>>.
 
 By default, the link text used in the generated TOC is
 based on the title of each file. You can provide an
-abbreviated title by using the DocBook `<titleabbrev>`
-tag. For example:
+abbreviated title by using the using a `titleabbrev` in
```
Comment (jrodewig) [inline README.asciidoc]:
```suggestion
abbreviated title using a `titleabbrev` in
```

File: resources/asciidoctor/lib/docbook_compat/titleabbrev_handler.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require_relative '../scaffold'
+
+module DocbookCompat
+  ##
+  # Looks for pass blocks with `<titleabbrev>` and adds an attributes to their
```
Comment (jrodewig) [inline resources/asciidoctor/lib/docbook_compat/titleabbrev_handler.rb]:
```suggestion
  # Looks for pass blocks with `<titleabbrev>` and adds an attribute to their
```

File: resources/asciidoctor/lib/docbook_compat/titleabbrev_handler.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  # parent section that is an abbreviated title usesd when rendering the
```
Comment (jrodewig) [inline resources/asciidoctor/lib/docbook_compat/titleabbrev_handler.rb]:
```suggestion
  # parent section. This attribute is an abbreviated title used when rendering the
```

File: resources/asciidoctor/lib/docbook_compat/titleabbrev_handler.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  # when rendering the table of contents. This exists entirely for backwards
```
Comment (jrodewig) [inline resources/asciidoctor/lib/docbook_compat/titleabbrev_handler.rb]:
```suggestion
  # the table of contents. This exists entirely for backwards
```

**Comment by nik9000** [comment]
> LGTM. Left a few nit comments, but none of those are blocking. Thanks!

Thanks!

**Pull request merged**